content/v2.0-preview/get-started/get-started-with-composition.md

```diff
@@ -3,4 +3,686 @@ title: Get Started With Composition
+          # if observed_deployment?.status?.availableReplicas:
+          #   _desired_xr.status.replicas = observed_deployment.status.availableReplicas
+
+
+          _desired_service = {
+            apiVersion = "v1"
+            kind = "Service"
+            metadata = {
+              annotations = {
+                "krm.kcl.dev/composition-resource-name" = "service"
+                "krm.kcl.dev/ready" = "True"
+              }
+              labels = {"example.crossplane.io/app" = observed_xr.metadata.name}
+            }
+            spec = {
+              selector = {"example.crossplane.io/app" = observed_xr.metadata.name}
+              ports = [{protocol = "TCP", port = 8080, targetPort = 80}]
+            }
+          }
+
+          observed_service = option("params").ocds["service"]?.Resource
+          if observed_service?.spec?.clusterIP:
+            _desired_service.metadata.annotations["krm.kcl.dev/ready"] = "True"
+            
+            # TODO(negz): Fix this.
+            # _desired_xr.status.address = observed_service.spec.clusterIP
+
+          items = [_desired_deployment, _desired_service, _desired_xr]
+```
+{{< /tab >}}
+
+{{< tab "Templated YAML" >}}
+Create this composition to use templated YAML to configure Crossplane:
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: app-templated-yaml
+spec:
+  compositeTypeRef:
+    apiVersion: example.crossplane.io/v1
+    kind: App
+  mode: Pipeline
+  pipeline:
+  - step: create-deployment-and-service
+    functionRef:
+      name: crossplane-contrib-function-go-templating
+    input:
+      apiVersion: ???
+      kind: ???
+      # TODO
+```
+{{< /tab >}}
+
+{{</ tabs >}}
+
+Save the composition as `composition.yaml` and apply it:
+
+```shell
+kubectl apply -f composition.yaml
+```
+
+{{<hint "note">}}
+A composition can include multiple functions.
+
+Functions can change the results of earlier functions in the pipeline.
+Crossplane uses the result returned by the last function.
+{{</hint>}}
+
+## Use the custom resource
+
+Crossplane now understands `App` custom resources.
+
+Create an `App`:
+
+```yaml
+apiVersion: example.crossplane.io/v1
+kind: App
+metadata:
+  namespace: default
+  name: my-app
+spec:
+  image: nginx
+```
+
+Save the `App` as `app.yaml` and apply it:
+
+```shell
+kubectl apply -f app.yaml
+```
+
+Check that the `App` is ready:
+
+```shell {copy-lines="1"}
+kubectl get -f app.yaml
+NAME     SYNCED   READY   COMPOSITION   AGE
+my-app   True     True    app-yaml      56s
+```
+
+{{<hint "note">}}
+The `COMPOSITION` column shows what composition the `App` is using.
+
+You can create multiple compositions for each kind of XR.
+[Read the XR page]({{<ref "../concepts/composite-resources">}}) to learn how to
+select which composition Crossplane uses.
+{{</hint>}}
+
+Check that Crossplane created a `Deployment` and a `Service`:
+
+```shell {copy-lines="1"}
+kubectl get deploy,service
```

Comment:
should we use `-l example.crossplane.io/app=my-app` here also, so only the resources for the app are shown? i had an extra service show up in my output that confused me for a second
```
❯ kubectl get deploy,service -l example.crossplane.io/app=my-app
NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/my-app-6qdhp   2/2     2            2           3m42s

NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/my-app-8tprb   ClusterIP   10.96.174.163   <none>        8080/TCP   3m42s
```

Reply:
Good idea - done.